modules/rustbitcoin/rust_bitcoin_lib/src/lib.rs

```diff
@@ -40,13 +40,17 @@ pub unsafe extern "C" fn rust_bitcoin_des_block(
     match res {
         Ok(res) => {
             let check = res.0.check_merkle_root() && res.0.check_witness_commitment();
-            return str_to_c_string(check.to_string().as_str());
+            if check {
+                return str_to_c_string(&res.0.block_hash().to_string());
+            } else {
+                return str_to_c_string("");
```

Comment:
```suggestion
                return str_to_c_string("0");
```
This way we can remove the empty check later, since in the end we just want "0" for comparison.